src/main/java/io/carbynestack/cli/client/castor/util/TupleFileParser.java

```diff
@@ -35,12 +37,36 @@ public static TupleChunk parse(TupleType tupleType, File tupleFile, UUID chunkId
 
     byte[] bytes;
     try (FileInputStream fileInputStream = new FileInputStream(tupleFile)) {
-      bytes = toByteArray(fileInputStream);
+      final byte[] fileBytes = toByteArray(fileInputStream);
+      bytes = removeTupleHeader(fileBytes);
     } catch (IOException e) {
       log.error(String.format("Cannot read file %s", tupleFile.getPath()));
       throw new RuntimeException();
     }
 
     return TupleChunk.of(tupleType.getTupleCls(), tupleType.getField(), chunkId, bytes);
   }
+
+  /**
+   * Removes the File Header that was introduced in MP-SPDZ v0.2.8
+   *
+   * @param tuplesWithHeader The file bytes that include the header
+   * @return The file bytes shifted by the header
+   */
+  private static byte[] removeTupleHeader(byte[] tuplesWithHeader) {
+
+    // See https://github.com/data61/MP-SPDZ/issues/418#issuecomment-975424591 for header
+    // Starts with LongEndian 8 Byte Number depicting the header length, followed by [headerLength]
+    // bytes
+    // ToDo: It would be possible to check the prime here, so that no incorrect values are uploaded,
+    // but Castor has no access to it.
```

Comment:
add indent to all subsequent ToDo lines e.g. to support highlighting in IDEs like IntelliJ (see [Multiline TODO comments](https://www.jetbrains.com/idea/whatsnew/2018-3/#editor))

Reply:
fixed